### ISSUE TITLE
Documentation missing (?) : indexing strategy

### ISSUE THREAD

**Issue by BenTalagan** [jetrockets/metka#31]
Hi, the docs do not give any clues about how to index tags columns for `metka`.

By default, building an index on a string column in a rails migration with `add_index` will create a standard `btree` index which is probably not what you want to use `metka` efficiently for querying.

Shouldn't a notice be added thus in the doc regarding indexing with `gist` and `gin` ?

```
add_index  :songs, :tags, using: 'gin'
```

Thanks for the awesome work, by the way!

**Comment by m0rozov** [comment]
Hello, @BenTalagan!

That's a good idea 👍 
We'll think about how to describe it correctly.

If you have time, we will be glad to have your pull-request 😉

**Comment by BenTalagan** [comment]
Thanks @m0rozov !

Actually, I'm having problems making an index work correctly with `metka`. From what I've found so far, an index declared as : 

```
add_index  :devices, :tags, using: 'gin'
```

will not be used by rails/metka/postgres (tested with `Model.with_any_tags('foo').explain`)

That might be related to the way the query is performed and casted. If I write my own scope, like this : 

```ruby
class Device
  scope :with_any_tag, -> (tags){ where('tags && ARRAY[?]::varchar[]', tags) }
end
```

I can get my index to be used properly (without `metka`).

```
Device.with_any_tag("foo").explain
D, [2020-11-19T13:39:24.011326 #70580] DEBUG -- :   Device Load (2.4ms)  SELECT "devices".* FROM "devices" WHERE (tags && ARRAY['foo']::varchar[])

EXPLAIN for: SELECT "devices".* FROM "devices" WHERE (tags && ARRAY['foo']::varchar[])
                                      QUERY PLAN
---------------------------------------------------------------------------------------
 Bitmap Heap Scan on devices  (cost=25.84..719.48 rows=237 width=632)
   Recheck Cond: (tags && '{foo}'::character varying[])
   ->  Bitmap Index Scan on index_devices_on_tags  (cost=0.00..25.78 rows=237 width=0)
         Index Cond: (tags && '{foo}'::character varying[])
(4 rows)
```

But with the `with_any_tags` querier from `metka` it won't be used. 

```
Device.with_any_tags("foo").explain
D, [2020-11-19T13:39:18.754687 #70580] DEBUG -- :   Device Load (53.5ms)  SELECT "devices".* FROM "devices" WHERE (CAST("devices"."tags" AS text[]) && CAST(ARRAY['foo'] AS text[]))

EXPLAIN for: SELECT "devices".* FROM "devices" WHERE (CAST("devices"."tags" AS text[]) && CAST(ARRAY['foo'] AS text[]))
                          QUERY PLAN
--------------------------------------------------------------
 Seq Scan on devices  (cost=0.00..2876.07 rows=237 width=632)
   Filter: ((tags)::text[] && '{foo}'::text[])
(2 rows)
```

I have taken the `::varchar[]` syntax directly from [the official rails guide].(https://guides.rubyonrails.org/active_record_postgresql.html#array)

Maybe worth double checking the implementation in `query_builder/base_query.rb` ? Do you have any opinion on this?

**Comment by igor-alexandrov** [comment]
@BenTalagan thanks for pointing this out!
We released 2.2.0 today with various improvements and fixes.

**Comment by BenTalagan** [comment]
Awesome, thanks!
